corehq/apps/app_manager/tests/test_suite_case_tiles.py

```diff
@@ -453,3 +494,61 @@ def test_case_tile_with_case_search(self, *args):
             # action[1] is the reg from case list action hard-coded into the default template
             "detail[@id='m0_search_short']/action[2]",
         )
+
+    def test_case_tile_with_sorting(self, *args):
+        factory = AppFactory()
+        module, form = factory.new_basic_module("my_module", "person")
+        module.case_details.short.case_tile_template = CaseTileTemplates.ONE_ONE_TWO.value
+        module.case_details.short.display = 'short'
+        self._add_columns_for_one_one_two_case_details(module)
+        sort_elements = [
+            SortElement(field='b', direction='ascending', type='plain'),
+            SortElement(field='a', direction='ascending', type='plain')
+        ]
+        module.case_details.short.sort_elements.extend(sort_elements)
+        suite = factory.app.create_suite()
+
+        template_xpath = './detail[@id="m0_case_short"]/field[1]/sort'
+        self.assertXmlPartialEqual(
+            """
+            <partial>
+                <sort direction="ascending" order="2" type="string">
+                    <text>
+                        <xpath function="a"/>
+                    </text>
+                </sort>
+            </partial>
+            """,
+            suite,
+            template_xpath,
+        )
+
+        template_xpath = './detail[@id="m0_case_short"]/field[2]/sort'
+        self.assertXmlPartialEqual(
+            """
+            <partial>
+                <sort direction="ascending" order="1" type="string">
+                    <text>
+                        <xpath function="b"/>
+                    </text>
+                </sort>
+            </partial>
+            """,
+            suite,
+            template_xpath,
+        )
+
+        template_xpath = './detail[@id="m0_case_short"]/field[4]/sort'
+        self.assertXmlPartialEqual(
+            """
+            <partial>
+                <sort type="string">
+                    <text>
+                        <xpath function="d"/>
+                    </text>
+                </sort>
+            </partial>
+            """,
+            suite,
+            template_xpath,
+        )
```

Comment:
"a" and "b" have sort blocks because a SortElement pointing to those fields are added module.case_details.short.sort_elements. https://github.com/dimagi/commcare-hq/blob/17c70cca6e6211d6c6fda361067fe81e164b41bc/corehq/apps/app_manager/tests/test_suite_case_tiles.py#L527-L528

"d" i'm honestly not sure why it has a sort block, but it seems to be tied to the format. Certain field formats, such as date, adds a sort block for a regular case list so to just carried that behavior over to case tiles.